localstack/utils/docker.py

```diff
@@ -596,5 +803,559 @@ def mountable_tmp_file():
+                params = {"stdout": 1, "stderr": 1, "stream": 1}
+                if interactive:
+                    params["stdin"] = 1
+                sock = container.attach_socket(params=params)
+                sock = sock._sock if hasattr(sock, "_sock") else sock
+                container.start()
+                with sock:
+                    try:
+                        if stdin:
+                            sock.sendall(to_bytes(stdin))
+                            sock.shutdown(socket.SHUT_WR)
+                        stdout, stderr = self._read_from_sock(sock, False)
+                    except socket.timeout:
+                        LOG.debug("timeout")
+                        pass
+                try:
+                    exit_code = container.wait()["StatusCode"]
+                    if exit_code:
+                        raise ContainerException(
+                            "Docker container returned with exit code %s" % exit_code,
+                            stdout=stdout,
+                            stderr=stderr,
+                        )
+                except APIError:
+                    pass
+            else:
+                container.start()
+            return stdout, stderr
+        except NotFound:
+            raise NoSuchContainer(container_name_or_id)
+        except APIError:
+            raise ContainerException()
+
+    def create_container(
+        self,
+        image_name: str,
+        *,
+        name: Optional[str] = None,
+        entrypoint: Optional[str] = None,
+        remove: bool = False,
+        interactive: bool = False,
+        tty: bool = False,
+        detach: bool = False,
+        command: Optional[Union[List[str], str]] = None,
+        mount_volumes: Optional[List[Tuple[str, str]]] = None,
+        ports: Optional[PortMappings] = None,
+        env_vars: Optional[Dict[str, str]] = None,
+        user: Optional[str] = None,
+        cap_add: Optional[str] = None,
+        network: Optional[str] = None,
+        dns: Optional[str] = None,
+        additional_flags: Optional[str] = None,
+    ) -> str:
+        LOG.debug("Creating container with image: %s", image_name)
+        if additional_flags:
+            env_vars, ports, mount_volumes = self._parse_additional_flags(
+                additional_flags, env_vars, ports, mount_volumes
+            )
+        try:
+            kwargs = {}
+            if cap_add:
+                kwargs["cap_add"] = [cap_add]
+            if dns:
+                kwargs["dns"] = [dns]
+            if ports:
+                kwargs["ports"] = ports.to_dict()
+            mounts = None
+            if mount_volumes:
+                mounts = dict(
+                    map(
+                        lambda paths: (str(paths[0]), {"bind": paths[1], "mode": "rw"}),
+                        mount_volumes,
+                    )
+                )
+
+            try:
+                container = self.client().containers.create(
```

Comment:
nit: We could put lines 1214-1228 into a small inline function, to reduce duplication with lines 1231-1245.